### PR TITLE
MultiNode compaction test

### PR DESCRIPTION
**Pull request by chessai**

### CODE REVIEW

File: src/Chainweb/Chainweb.hs [hunk 1]
```diff
@@ -328,9 +328,9 @@ validatingMempoolConfig cid v gl gp mv = Mempool.InMemConfig
         f (This _) = Left (T2 (Mempool.TransactionHash "") (Mempool.InsertErrorOther "preInsertBatch: align mismatch 1"))
 
 
-data StartedChainweb logger
-    = forall cas. (CanReadablePayloadCas cas, Logger logger) => StartedChainweb !(Chainweb logger cas)
-    | Replayed !Cut !Cut
+data StartedChainweb logger where
```
Comment (chessai) [inline src/Chainweb/Chainweb.hs:331]: This was easier to read in GADT syntax

File: src/Chainweb/Pact/Backend/Compaction.hs [hunk 2]
```diff
@@ -164,35 +164,64 @@ runCompactM e a = runReaderT (unCompactM a) e
 
 -- | Prepare/Execute a "$VTABLE$"-templated query.
 execM_ :: ()
-  => Text -- ^ "$VTABLE$"-templated query
```
Comment (chessai) [inline src/Chainweb/Pact/Backend/Compaction.hs:167]: Added query logging to all of this; this should become more like chainweb logging, though. That will be in a separate PR

**Review by edmundnoble — APPROVED** [review]

File: test/Chainweb/Test/MultiNode.hs [hunk 3]
```diff
@@ -283,6 +288,62 @@ runNodesForSeconds loglevel write baseConf n (Seconds seconds) rdb pactDbDir inn
...
+  -> TestTree
+compactAndResumeTest logLevel v n =
+  let
+    name = "compact-resume"
+  in
+  after AllFinish "ConsensusNetwork" $ testCaseSteps name $ \step ->
+  withTempRocksDb "compact-resume-test-rocks" $ \rdb ->
+  withSystemTempDirectory "compact-resume-test-pact" $ \pactDbDir -> do
+    let tastyLog = step . T.unpack
+    let logFun = tastyLog
```
[the remark this answers is not in the document]
Comment (chessai) [reply test/Chainweb/Test/MultiNode.hs:311]: Was copying from below (`replayTest`). I can see the argument for it, because it differentiates between what is a test log msg from us the programmer in the test itself, vs something transitively called by our code.

However, I also am somewhat inclined to remove it.
